server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -475,7 +475,10 @@ public synchronized void open(String address) throws IOException {
       throw new IOException(ex);
     }
 
-    syncThread = Threads.createThread("Accumulo WALog thread " + this, new LogSyncingTask());
+    // TODO KEVIN RATHBUN this seems like a vital thread for TabletServer, but appears that the
+    // thread will continuously be recreated, so probably fine to stay non critical
+    syncThread =
+        Threads.createNonCriticalThread("Accumulo WALog thread " + this, new LogSyncingTask());
```

Comment:
this seems like a vital thread for TabletServer, but appears that the thread will continuously be recreated, so probably fine to stay non critical

Reply:
The syncThread is created for each DFSLogger, it looks like there is a 1:1 relationship. The syncThread runs a LogSyncingTask which flushes the write ahead log according to the users configured durability intention. When the write-ahead logs are full, the TabletServer will create new ones and therefore new sync threads. I don't think a sync thread for a specific DFSLogger is recreated, which means that it's possible it could die. I think this might be critical.

Reply:
Yeah this should probably be critical.  I suspect if it dies that writes to a tserver could freeze indefinitely.

Reply:
Addressed in 2c4ce6bc57b166a2a57eb61c11df5b2bb8da76b8